noq_django/rest_api/api/user_api.py

```diff
@@ -59,29 +60,35 @@ def list_available(request, selected_date: str):
     
     return [{"host": host, "products": products} for host, products in hostproduct_dict.items()]
 
-@router.post("/request_booking", response=BookingSchema, tags=["user-booking"])
-def request_booking(request, booking_data: BookingPostSchema):
-
+@router.post("/booking", response=BookingSchema, tags=["user-booking"])
+def create_booking(request, booking_data: BookingPostSchema):
     try:
         product = Product.objects.get(id=booking_data.product_id)
-        Available.objects.filter(product=product)
-    except Available.DoesNotExist:
-        raise HttpError(404, "Product is not available")
+        available = Available.objects.filter(product=product, available_date__range=[booking_data.start_date, booking_data.end_date]).exists()
```

Comment:
This will not work. If there is no bookings for a specific day, there will not be data row in Available for that product and date. Available row is added only after someone as booked a product.

Reply:
Check should be: Not in the available or (for each day) places_left > 0.